### PR TITLE
:bug:issue-1711: Dependency on deprecated github.com/golang/mock

### PR DESCRIPTION
**Pull request by MykolaRodin**
**What this PR does / why we need it**:

This MR uses `uber-go/mock` instead of deprecated `github.com/golang/mock/gomock`

Fixes: #1711

### CODE REVIEW

**Comment by mdbooth** [comment]
/ok-to-test

Thank you! I'll update the description with a `Fixes` stanza so #1711 is automatically closed when this merges.

File: go.mod [hunk 1]
```diff
@@ -5,17 +5,17 @@ go 1.22
 require (
 	github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc
 	github.com/go-logr/logr v1.4.1
-	github.com/golang/mock v1.6.0
 	github.com/google/go-cmp v0.6.0
 	github.com/google/gofuzz v1.2.0
 	github.com/google/uuid v1.4.0
-	github.com/gophercloud/gophercloud v1.11.0
+	github.com/gophercloud/gophercloud v1.7.0
```
Comment (lentzi90) [inline go.mod:11]: This seems unrelated and unnecessary. Please keep it as it was

Comment (MykolaRodin) [reply go.mod]: `github.com/gophercloud/gophercloud v1.11.0` has been restored

**Comment by lentzi90** [comment]
Thanks! May I also ask that you squash the commits to one, please!

**Review by lentzi90 — COMMENTED** [review]
Thanks!
/lgtm

**Comment by MykolaRodin** [comment]
I tried to do it using `git rebase -i` with subsequent steps and failed. That is why I deleted my last commit (`git reset --hard 1c5dc315bb125f92f43778bf1a75b6ad05f72739` and `git push --force`), made the required modifications in the code and committed them modifying the only remaining commit (`git commit --amend` and `git push --force`). That is the best I can do.

**Comment by lentzi90** [comment]
It looks fine :slightly_smiling_face:

**Comment by MykolaRodin** [comment]
Thank you. If it is fine, could you please resolve our conversation above?

**Comment by lentzi90** [comment]
I have already done so. This will merge as soon as all the tests have passed. The e2e test takes about 1 h so I estimate that this will be merge in about half an hour. :slightly_smiling_face:
